### PR TITLE
tech-story: [M3-8873] - Upgrade to storybook 8.4.2

### PR DESCRIPTION
**Pull request by abailly-akamai**
## Description 📝
Update storybook to 8.4.x to get the latest dependency cleanup benefits 

## Changes  🔄
List any change relevant to the reviewer.
- Upgrade to storybook 8.4.2
- Fix display of extended props (allow React DocGen to parse UI package)

Confirm all works properly but running locally:
- `yarn storybook`
- `yarn build-storybook`

### CODE REVIEW

**Comment by bnussman-akamai** [comment]
Not related to this PR, but auto-docs is not working **on develop** for me 🫤 Do you see the same?

|  develop | design.linode.com |
|---|---|
|  Screenshot 2024-11-07 at 1 06 50 PM | Screenshot 2024-11-07 at 1 06 44 PM  |

**Comment by abailly-akamai** [comment]
@bnussman-akamai this should be fixed

File: packages/manager/.storybook/main.ts [hunk 1]
```diff
@@ -11,18 +12,18 @@ const config: StorybookConfig = {
...
   addons: [
-    '@storybook/addon-docs',
-    '@storybook/addon-controls',
-    '@storybook/addon-viewport',
-    '@storybook/addon-measure',
-    '@storybook/addon-actions',
-    'storybook-dark-mode',
-    '@storybook/addon-storysource',
-    '@storybook/addon-a11y',
+    getAbsolutePath('@storybook/addon-docs'),
```
Comment (bnussman-akamai) [inline packages/manager/.storybook/main.ts:15]: What's the reason for this change? Seems to work fine just passing the plugin name

**Review by bnussman-akamai — APPROVED** [review]

File: packages/manager/.storybook/main.ts (hunk 1 above)
Comment (abailly-akamai) [reply packages/manager/.storybook/main.ts:15]: Screenshot 2024-11-07 at 13 52 13

This was the recommendation from the `npx` upgrade script, which the codemod modified - i don't really see much harm in it either way

Screenshot 2024-11-07 at 13 52 13

Comment (bnussman-akamai) [reply packages/manager/.storybook/main.ts:15]: Makes sense. Thanks!

File: packages/manager/.storybook/utils.ts [hunk 2]
```diff
@@ -13,6 +13,9 @@ export const getReactDocgenTSFileGlobs = () => {
   const filesWithStories = globby.sync(PATTERN);
   const files = new Set<string>();
 
+  files.add('**/*.{ts,tsx}');
+  files.add('../ui/**/*.{ts,tsx}');
```
Comment (hkhalil-akamai) [inline packages/manager/.storybook/utils.ts:17]: Doing this is equivalent to not having this optimization at all (which is supposed to find only TS files with associated stories). 

The real issue lies on line 3, which I broke in [#11184](https://github.com/linode/manager/pull/11184/files#diff-df54c0a4b121f9abeb3a02c5e3960a3f588830de910f72b8cff38db93e43943b). I modified it to use absolute paths since relative paths failed the unit test when run from the repo root. Unfortunately, it seems like absolute paths are not supported by storybook.

At this point, our two options are:

1. Remove this optimization from `main.ts` altogether
2. Find a fix for that still allows tests to pass when run from repo root
3. Remove the test and go back to relative paths

Comment (abailly-akamai) [reply packages/manager/.storybook/utils.ts:17]: i'll take a better look and see if it really affects start time, cause maybe we don't really need it

Comment (bnussman-akamai) [reply packages/manager/.storybook/utils.ts:17]: I was kind of iffy on the optimization in the first place https://github.com/linode/manager/pull/10762#pullrequestreview-2248444669

I'm okay with removing it if we deem it isn't crucial

Comment (abailly-akamai) [reply packages/manager/.storybook/utils.ts]: I spend a bit of time cold starting Storybook with and without the util and I have not found any performance benefit. I removed it in this PR to avoid benefit-less extra complexity.

All that being said I do agree performance could be improved - I tried a few things but apart from weird ways of hacking what we parse I also think it's best to keep our config close to the defaults for now and hope for some of the many improvements Storybook to get us to a better place. There's certainly a risk a complex config could obfuscate a core improvement from the platform. CC @hana-akamai 

The only reason we use `react-docgen-typescript` is to get extended types from MUI to populate our prop table, and this is the main offender. If we can find a way to do this without this `reactDocgen` we'd see improvements across the board. 

Finally, I do wonder if this PR brings up some speed to the cold start, considering the dependency updates. It does seem a tad faster to me but I could be blinded by my disdain about how bloated Storybook always has been and happy to see it slimming down.

Comment (hana-akamai) [reply packages/manager/.storybook/utils.ts]: @abailly-akamai the performance improvement is more noticeable on older machines (intel chip) but I'm fine with removing it if the team feels it's more trouble than it's worth

I'm on a refurbished M1 now so the performance improvement isn't that noticeable

**Comment by abailly-akamai** [comment]
There are a few more issues I have discovered with this update which I will want to spend more time on in the near future, We may have to wait for a couple more version updates to bump to 8.4.x

closing this for now

**Comment by hkhalil-akamai** [comment]
@abailly-akamai interested to know what you found -- I just glanced over it briefly but couldn't find any issues

**Comment by abailly-akamai** [comment]
@hkhalil-akamai when starting storybook, look at your console, you'll see 404s for some scripts when storybook opens in a new window (or reloads from a previous session). This actually blocks the stories from loading.

If you refresh manually it will work, which seems to indicate the window opens before some bundles have loaded.

It's pretty disruptive and after trying a bunch of fixes to get around it I feel it is better to not affect the experience too much right now considering the current `@linode/ui` migration. I'll revisit this soon - meanwhile we do need to fix the prop table with imports (without the storybook update) which you can do in any upcoming migration PR

**Comment by hkhalil-akamai** [comment]
Thanks for that context.

> meanwhile we do need to fix the prop table with imports (without the storybook update)

Created M3-8904 to handle this and will open a separate PR to facilitate discussion.
